### ISSUE TITLE
The WMS driver seems to consider that BBOX is always in WGS84

### ISSUE THREAD

**Issue by mmomtchev** [OSGeo/gdal#4093]
## Expected behavior and actual behavior.

The WMS driver seems to consider that BBOX is always in WGS84.
According to the WMS specification, the BBOX parameter should be in the CRS of the layer.

*The mandatory BBOX parameter allows a client to request a particular Bounding Box. Bounding Boxes are
defined in 6.7.4. The value of the BBOX parameter in a GetMap request is a list of comma-separated real
numbers (see 6.5) in the form "minx,miny,maxx,maxy". These values specify the minimum X, minimum Y,
maximum X, and maximum Y values of a region in the Layer CRS of the request. The units, ordering and direction
of increment of the x and y axes are as defined by the Layer CRS (see 6.7.3)*

It is not very clear if it should always be in the main CRS or the currently requested CRS (seems more logical), but it should not always be in WGS84 for sure

## Steps to reproduce the problem.

You don't need a WMS server, this operation fails before the WMS server is accessed
```
gdal_translate -tr 1 1  "WMS:http://test/wms?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&LAYERS=some:layer&SRS=EPSG:3857&BBOX=4230140,-893463,5899933,1345708" out.png
```

it will fail with
```
Input file size is 800709986, 1073741824
Warning 1: libpng: Image width exceeds user limit in IHDR
Warning 1: libpng: Image height exceeds user limit in IHDR
ERROR 1: libpng: Invalid IHDR data
```

## Operating system

Ubuntu 20.04

## GDAL version and provenance

custom GDAL 3.2.2 build (but this part hasn't changed in years)

**Comment by mmomtchev** [comment]
It is only the default `bbox`, not all `bbox`, my resolution was wrong in the first example

**Comment by jratike80** [comment]
Could you explain what is the default bbox for GDAL? Does it read the GetCapabilities, and use the LatLonBoundingBox (WMS 1.1.x) or EX_GeographicBoundingBox (WMS 1.3.0) of the selected layer, re-projected into the SRS/CRS used in the GetMap?

**Comment by mmomtchev** [comment]
@jratike80, that would be great, but it requires substantial modifications to the current architecture - the WMS driver considers that all metadata is contained in the URL
This is the reason most of the processing happens even if the server does not exist - the connection is established only when the raster data is needed

**Comment by jratike80** [comment]
I was reading the documentation https://gdal.org/drivers/raster/wms.html and made a test with

`gdalinfo "WMS:https://demo.geo-solutions.it/geoserver/wms"`

I see subdatasets with BBOX that must come from GetCapabilities.

`SUBDATASET_34_NAME=WMS:https://demo.geo-solutions.it:443/geoserver/wms?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&LAYERS=test%3Athematism_province&SRS=EPSG:32632&BBOX=166021.44309607695,0.0,833978.5569039229,9329005.182450451`

It seems like you used the URL type "a pseudo GetMap request, such as the subdataset name returned by the previous syntax" but without the BBOX.

**Comment by mmomtchev** [comment]
Yes, `gdalinfo` constructs URLs with the `bbox` argument - which is mandatory by the way - but the WMS driver will accept an URL without `bbox` and will add, in this case, the hard-coded string `"bbox=-90,-180,90,180"`

**Comment by jratike80** [comment]
Hard coded BBOX does not make sense to me. It is quite impossible to guess a reasonable bbox for a layer without reading GetCapabilities or input from a well-informed user. Missing bbox could rather throw an error.

**Comment by mmomtchev** [comment]
That would be in line with the WMS specifications, but it will surely break some existing applications
Normally, my opinion is that specifications, be they good or bad, are to be implemented by the letter, because otherwise you contribute to undermining them - especially in such a widely used library as GDAL

**Comment by jratike80** [comment]
I am not sure if it would break anything. I cannot imagine any application to do anything useful by letting GDAL to add a default bbox that covers the whole world.

**Comment by mmomtchev** [comment]
If the layer uses EPSG:4326, this would work with lots of WMS servers - some will return a valid file even if the layer does not cover the whole world
But I agree, throwing an error seems the right decision to me

**Comment by mmomtchev** [comment]
@jratike80 According to the specifications, the `CRS` argument is also mandatory - the WMS driver will add `EPSG:4326` if none is specified - and so is the `STYLE` - which the WMS driver does not add if none is specified. Lots of WMS implementations skip the `STYLE` which is mandatory for the client and optional for the server.
I don't think that removing the default value for `CRS` is a very good idea - I am quite confident that there are people that depend on it.
